BudgetTracker.BudgetSquirrel.WebApi/Controllers/AuthenticationApiController.cs

```diff
@@ -64,55 +56,46 @@ public async Task<IActionResult> DeleteUser(ApiRequest request)
         }
 
         /// <summary>
-        /// Returns a new token with claims information for the user specified
-        /// in the request. This token should be attached as a bearer token in
-        /// sub-sequent requests that require authentication.
+        /// Returns a new cookie with claims information for the user specified
+        /// in the request. This cookie will persistent until the user logs out
+        /// or the session times out.
         /// </summary>
-        /// <param name="data">
-        /// A JSON object that contains 2 keys:
-        /// - username
-        /// - password
-        /// These are used to authenticate the user for which this token is generated.
-        /// </param>
+        /// <param name="credentials">The username password used for the attempt to login <see cref="Credentials"/></param>
         /// <returns>
-        /// A JWT Token with claims information for the authenticated user. If
+        /// A Cookie with claims information for the authenticated user. If
         /// the user cannot be authenticated, a 403 will be thrown.
         /// </returns>
         [HttpPost("authenticate")]
-        public async Task<IActionResult> Authenticate(Dictionary<string, string> data)
+        public async Task<IActionResult> Authenticate(Credentials credentials)
         {
             User authenticatedUser;
             UserResponseMessage userResponse;
             try
             {
-                authenticatedUser = await _authenticationService.Authenticate(data["username"], data["password"]);
+                authenticatedUser = await _authenticationService.Authenticate(credentials.Username, credentials.Password);
                 userResponse = new UserResponseMessage(authenticatedUser);
             }
             catch (AuthenticationException e)
             {
-                return new JsonResult(new {
+                return this.BadRequest(new JsonResult(new {
                     error = e.Message
-                });
+                }));
             }
 
-            string securityKey = _authConfig.JWTSecurityKey;
-            SymmetricSecurityKey symSecKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(securityKey));
-            SigningCredentials creds = new SigningCredentials(symSecKey, SecurityAlgorithms.HmacSha256);
+            if (authenticatedUser != null)
+            {
+                var cliamsIdenity = new ClaimsIdentity(authenticatedUser.CreateUserClaims(), CookieAuthenticationDefaults.AuthenticationScheme);
+
+                await HttpContext.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme,
+                    new ClaimsPrincipal(cliamsIdenity),
+                    new AuthenticationProperties { IsPersistent = true});
+
+                return this.Ok(authenticatedUser);
```

Comment:
Return the userResponse. That is the response message for the API. It only includes certain fields. For example, it doesn't include password whereas the User model includes the password.

This is a pattern I'd like to keep throughout the app. Everything returned in the API should be a separate class that represents a response message. The domain models should never be returned directly by an API call. I realize we aren't consistent with that currently but I'd like to be. That way we have a separation of concerns with models/messages.

messages represent a response that contains data from the domain layer.

Another PRO of this pattern is that the API layer is separate from the Domain layer. If we want to return an combination of 2 domain models in an API response, we just define the message class for it with the right data, create the logic to convert from Domain objects to Message instance and voila! we have our response. That way we don't have to modify models in the domain layer.

Reply:
Yeah whoops didn't realize we were passing the password to the front end. I will update it to `UserResponse`